### PR TITLE
Add line numbers with anchoring in log viewer

### PR DESCRIPTION
**Pull request by Martchus**
Related ticket: https://progress.opensuse.org/issues/165402

---

screenshot_20240820_173655

---

screenshot_20240820_173927

### CODE REVIEW

**Comment by perlpunk** [comment]
Can you change it so it does not select the line numbers when selecting the text?

**Comment by Martchus** [comment]
Note that I changed the filter to use query parameters. I mainly did this so it doesn't interfere with the normal anchoring behavior of the web browser that is now utilized to jump to log lines. Otherwise I would have to parse the fragment in JavaScript and implement the jumping manually. However, I don't think this is a bad change. In fact, it is more consistent with other places where we provide filtering parameters. (And I could actually reduce the required code by simply reusing helper functions we already have.)

**Comment by Martchus** [comment]
> Can you change it so it does not select the line numbers when selecting the text?

It does *not* select the line numbers when selecting text. It only selects the line numbers when clicking on one of the newly added line numbers on the left explicitly.

**Comment by perlpunk** [comment]
A minor bug:
* Select some line number
* Search for something else that doesn't appear in that line, so the selected line disappears as expected. Don't unfocus the search field
* Then select one of the filtered line numbers. It does not work. Only when you click it the second time

When unfocusing the search field first, it works.

**Comment by perlpunk** [comment]
Is it possible to select multiple line numbers?

**Comment by perlpunk** [comment]
> It does _not_ select the line numbers when selecting text.

It does select them for me.

**Comment by perlpunk** [comment]
log-view-line-numbers
Same behaviour in chromium and firefox.

**Comment by Martchus** [comment]
> A minor bug:

I'll fix that tomorrow.

> Is it possible to select multiple line numbers?

No. I guess implementing that would be a bit more work.

> It does select them for me.

I misunderstood. I thought you were talking about the hash/highlighting being updated. I'll change it tomorrow so line numbers are not selected :-)

File: assets/javascripts/test_result.js [hunk 1]
```diff
@@ -461,13 +461,16 @@ function activateTabAccordingToHashChange() {
...
+    link = $("[href='#details']");
+    tabName = 'details';
+  } else if (hash.startsWith('#step/')) {
     setCurrentPreviewFromStepLinkIfPossible(link);
     link = $("[href='#details']");
     tabName = 'details';
     // note: It is not a problem if the details haven't been loaded so far. Once the details become available the hash
     //       is checked again and the exact step preview will be shown.
-  } else if (hash.search('#comment-') === 0) {
+  } else if (hash.startsWith('#comment-')) {
```
Comment (b10n1k) [inline assets/javascripts/test_result.js:473]: Can this merged somehow?

**Comment by Martchus** [comment]
The latest version should fix the mentioned problems. I haven't implemented anchoring a range of lines because that would be a bit more involved.

**Review by perlpunk — APPROVED** [review]

**Review by kalikiana — APPROVED** [review]

**Review by b10n1k — APPROVED** [review]

**Pull request merged**